apps/omg_child_chain/lib/omg_child_chain/block_queue/block_submission.ex

```diff
@@ -0,0 +1,192 @@
+  # parity error code for duplicated tx
+  def process_result(
+        submission,
+        {:error, %{"code" => -32_010, "message" => "Transaction with the same hash was already imported."}},
+        _newest_mined_blknum
+      ) do
+    log_known_tx(submission)
+    :ok
+  end
+
+  def process_result(
+        submission,
+        {:error, %{"code" => -32_000, "message" => "replacement transaction underpriced"}},
+        _newest_mined_blknum
+      ) do
+    log_low_replacement_price(submission)
+    :ok
+  end
+
+  # parity version
+  def process_result(
+        submission,
+        {:error, %{"code" => -32_010, "message" => "Transaction gas price is too low. There is another" <> _}},
+        _newest_mined_blknum
+      ) do
+    log_low_replacement_price(submission)
+    :ok
+  end
+
+  def process_result(
+        submission,
+        {:error, %{"code" => -32_000, "message" => "authentication needed: password or unlock"}},
+        newest_mined_blknum
+      ) do
+    diagnostic = prepare_diagnostic(submission, newest_mined_blknum)
+    log_locked(diagnostic)
+    {:error, :account_locked}
+  end
+
+  def process_result(
+        submission,
+        {:error, %{"code" => -32_000, "message" => "nonce too low"}},
+        newest_mined_blknum
+      ) do
+    process_nonce_too_low(submission, newest_mined_blknum)
+  end
+
+  # parity specific error for nonce-too-low
+  def process_result(
+        submission,
+        {:error, %{"code" => -32_010, "message" => "Transaction nonce is too low." <> _}},
+        newest_mined_blknum
+      ) do
+    process_nonce_too_low(submission, newest_mined_blknum)
+  end
+
+  # ganache has this error, but these are valid nonce_too_low errors, that just don't make any sense
+  # `process_nonce_too_low/2` would mark it as a genuine failure and crash the BlockQueue :(
+  # however, everything seems to just work regardless, things get retried and mined eventually
+  # NOTE: we decide to degrade the severity to warn and continue, considering it's just `ganache`
+  def process_result(
+        _submission,
+        {:error, %{"code" => -32_000, "data" => %{"stack" => "n: the tx doesn't have the correct nonce" <> _}}} = error,
+        _newest_mined_blknum
+      ) do
+    log_ganache_nonce_too_low(error)
+    :ok
+  end
+
+  defp log_ganache_nonce_too_low(error) do
+    # runtime sanity check if we're actually running `ganache`, if we aren't and we're here, we must crash
+    :ganache = Application.fetch_env!(:omg_eth, :eth_node)
+    _ = Logger.warn(inspect(error))
+    :ok
+  end
+
+  defp log_success(submission, txhash) do
+    _ = Logger.info("Submitted #{inspect(submission)} at: #{inspect(txhash)}")
+    :ok
+  end
+
+  defp log_known_tx(submission) do
+    _ = Logger.debug("Submission #{inspect(submission)} is known transaction - ignored")
+    :ok
+  end
+
+  defp log_low_replacement_price(submission) do
+    _ = Logger.debug("Submission #{inspect(submission)} is known, but with higher price - ignored")
```

Comment:
can we not log debug 🙏 I have found it quite useless for real env investigation and if you just want sth for development you can add and remove those yourself locally.

Reply:
I would argue some of our logs are better kept as debug, but I agree we need these logs on block submissions

Reply:
Done! https://github.com/omgnetwork/elixir-omg/pull/1575/commits/791ab228bee6f821782dd5a43714b9762e9cff4a